### PR TITLE
Update checkout discount calculations

### PR DESCRIPTION
**Pull request by mateuszgrzyb**
I want to merge this change because...

I want to add discount calculations based on `include_taxes_in_prices` site setting
(SALEOR-4943)

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4943 Update checkout's discount calculations](https://app.clickup.com/t/2549495/SALEOR-4943)

**Comment by mateuszgrzyb** [comment]
Depends on #8103

**Review by fowczarek — APPROVED** [review]

File: saleor/checkout/utils.py [hunk 1]
```diff
@@ -713,3 +765,7 @@ def validate_variants_in_checkout_lines(lines: Iterable["CheckoutLineInfo"]):
                 )
             }
         )
+
+
+def get_base_money(taxed_money: "TaxedMoney", from_gross: bool) -> Money:
```
Comment (korycins) [inline saleor/checkout/utils.py:770]: Not sure if the name gives enough context. What is base_money? The function returns a money amount which we used in all calculations, based on the flag value

File: saleor/graphql/checkout/mutations.py [hunk 2]
```diff
@@ -654,8 +660,14 @@ def perform_mutation(cls, _root, info, line_id, checkout_id=None, token=None):
             checkout, lines, info.context.discounts, manager
         )
         update_checkout_shipping_method_if_invalid(checkout_info, lines)
+
+        taxes_included = include_taxes_in_prices()
```
Comment (korycins) [inline saleor/graphql/checkout/mutations.py:664]: `info.context.site.settings.include_taxes_in_prices`

**Review by korycins — APPROVED** [review]

**Pull request merged**